docs/pages/components/file-uploader.md

```diff
@@ -0,0 +1,143 @@
+---
+title: File Uploader
+id: file-uploader
+keywords: file uploader,files
+sidebar: left-navigation-sidebar
+toc: false
+permalink: components/file-uploader.html
+folder: components
+summary:
+tags: [f3, a11y, theme]
+---
+
+The File uploader component is an opinionated composition of the input and button components.
+Please see the `input` and `button` components documentation for the complete list of modifiers.
+<br/>
+<br/>
+`File Uploader` allows user to select the files when click on the input or browse button.
+Input element is readonly and populated with the selected file name.
+{: .docs-intro}
+<br/>
+
+## Cozy Mode
+
+{% capture default %}
+
+<div class="fd-form-item">
+    <div class="fd-file-uploader">
+        <input class="fd-input fd-file-uploader__input" 
+        onclick="browseFile('input1');" 
+        title="Select a file for uploading" 
+        aria-label="Select a file for uploading" 
+        type="text" 
+        id="browse_input1" 
+        placeholder=" Choose a file for upload to localserver " 
+        readonly>
+        <button class="fd-button fd-file-uploader__button" 
+        width="40px"
+        onclick="browseFile('input1');" 
+        id="fileuploader-button1" 
+        aria-label="Select a file for uploading" >
+        Browse...
+        </button>
+    </div>
+      <input
```

Comment:
That's an interesting approach, but I'm not sure if it will satisfy screen readers since visible input is `type="text"`.

@jbadan @jacobdevera @prsdthkr ?

Reply:
Right, this seems problematic due to the readonly input with `type="text"`. When read by a screenreader, it is a little confusing how the field should be interacted with. Ideally, we use a input `type="file"`, and it should only be one tab stop. Currently, you can tab to both the input and button.

Reply:
We can't use the input with type="file" because it does't match the Fiori 3s styles , so we are using the type="text" and button component.

Reply:
May i know why i shouldn't use readonly attribute, can you explain it clearly?

Reply:
one tab stop changes are implemented now.

Reply:
Can the styles and markup be adjusted to work with type="file"?

This UI5 example appears to be on the right track, and matches the Fiori 3 style.
https://openui5.hana.ondemand.com/entity/sap.ui.unified.FileUploader/sample/sap.ui.unified.sample.FileUploaderBasic

Readonly is not supported for a file input. I assume it is only being used to prevent the text input from being edited when a file is uploaded, but it is unneeded for a file input.